teku/src/main/java/tech/pegasys/teku/cli/subcommand/debug/DebugDbCommand.java

```diff
@@ -432,6 +435,244 @@ public int dumpBlobSidecars(
+            final Bytes32 hotBlockRoot = Bytes32.wrap(rootAndHotBlock.getKey());
+            final Optional<SignedBeaconBlock> hotBlock = database.getHotBlock(hotBlockRoot);
+            if (hotBlock.isPresent()) {
+              final int expectedBlobSidecarsCount =
+                  hotBlock
+                      .get()
+                      .getMessage()
+                      .getBody()
+                      .getOptionalBlobKzgCommitments()
+                      .map(SszList::size)
+                      .orElse(0);
+              for (int blobSidecarIndex = 0;
+                  blobSidecarIndex < expectedBlobSidecarsCount;
+                  blobSidecarIndex++) {
+                final SlotAndBlockRootAndBlobIndex blobSidecarKey =
+                    new SlotAndBlockRootAndBlobIndex(
+                        hotBlockSlot, hotBlockRoot, UInt64.valueOf(blobSidecarIndex));
+
+                // Checking blob sidecar existence
+                final Optional<BlobSidecar> maybeCurrentBlobSidecar =
+                    database.getBlobSidecar(blobSidecarKey);
+                if (maybeCurrentBlobSidecar.isEmpty()
+                    && hotBlockSlot.isGreaterThanOrEqualTo(lowerSlotBoundDataAvailability)) {
+                  // inside DA window but missing
+                  missingHotBlobsSidecars++;
+                  System.err.printf(
+                      "ERROR: Unable to locate hot blob sidecar %s%n", blobSidecarKey);
+                }
+              }
+            }
+          }
+        }
+      }
+
+      // Checking orphaned blob sidecars
+      try (final Stream<SlotAndBlockRootAndBlobIndex> blobsSidecarKeys =
+          database.streamBlobSidecarKeys(UInt64.ZERO, UInt64.MAX_VALUE)) {
+        for (final Iterator<SlotAndBlockRootAndBlobIndex> blobSidecarsIterator =
+                blobsSidecarKeys.iterator();
+            blobSidecarsIterator.hasNext(); ) {
+          final SlotAndBlockRootAndBlobIndex blobSidecarKey = blobSidecarsIterator.next();
+
+          sidecarsToBlocksCheckCounter++;
+
+          final Optional<SignedBeaconBlock> maybeFinalizedBlockFound =
+              database
+                  .getFinalizedBlockAtSlot(blobSidecarKey.getSlot())
+                  .filter(block -> block.getRoot().equals(blobSidecarKey.getBlockRoot()));
+
+          final Optional<SignedBeaconBlock> maybeHotBlockFound =
+              database.getHotBlock(blobSidecarKey.getBlockRoot());
+
+          if (maybeHotBlockFound.isEmpty() && maybeFinalizedBlockFound.isEmpty()) {
+            orphanBlobsSidecars++;
+            System.err.printf(
+                "ERROR: blob sidecar has no corresponding finalized or hot block %s%n",
+                blobSidecarKey);
+          }
+
+          // check Data Availability
+          if (blobSidecarKey.getSlot().isLessThan(lowerSlotBoundDataAvailability)) {
+            unprunedBlobsSidecars++;
+            System.err.printf("WARNING: non pruned blobs sidecar %s%n", blobSidecarKey);
```

Comment:
there could be a lot of them if `epochsStoreBlobs` override were used, not sure how to handle this

Reply:
that's a good point. An option could be to add a DA override param to the CLI to se the expected earliest slot

Reply:
makes sense! added an optional `--lower-data-availability-epoch` option